### PR TITLE
[LLM] Support mixed_fp8 on Arc

### PR DESCRIPTION
**Pull request by sgwhat**
## Description

Support `mixed_fp8` quantization on ARC.

### 1. Why the change?

Select the optimal 8-bit format on a layer-wise basis.

### 2. User API changes

```python
model = AutoModelForCausalLM.from_pretrained(model_path,
                                             load_in_low_bit="mixed_fp8",
                                             optimize_model=True,
                                             trust_remote_code=True,
                                             use_cache=True)
```

### 3. Summary of the change 

- Add `mixed_fp8` in `load_in_low_bit`.
- Unify shared logits across `mixed_fp4` and `mixed_fp8`.
- Rename `mixed_int4` to `mixed_fp4`.

### 4. How to test?
- [ ] Unit test

### CODE REVIEW

File: python/llm/src/bigdl/llm/transformers/low_bit_linear.py [hunk 1]
```diff
@@ -194,6 +195,26 @@ def __new__(cls,
...
+        from torch.nn.functional import mse_loss
+        w_quant_q = ggml_convert_qtype(w, SYM_INT_TYPE,
+                                       device=device,
+                                       convert_shape_only=self.convert_shape_only)
+        w_q_dequant = ggml_convert_fp32(w_quant_q, w.shape,
+                                        reduce(mul, w.shape, 1), SYM_INT_TYPE)
+        q_mse = mse_loss(w_q_dequant, w)
+        return q_mse, w_quant_q
+
+    def ggml_w_fq_mse(self, w, FP_TYPE, device):
```
Comment (hkvision) [inline python/llm/src/bigdl/llm/transformers/low_bit_linear.py:208]: aren't these two functions doing the same thing??

File: python/llm/src/bigdl/llm/transformers/model.py [hunk 2]
```diff
@@ -113,7 +113,7 @@ def load_convert(cls, q_k, optimize_model, *args, **kwargs):
         invalidInputError(q_k in ggml_tensor_qtype,
                           f"Unknown load_in_low_bit value: {q_k}, expected:"
                           f" sym_int4, asym_int4, sym_int5, asym_int5, sym_int8, nf3, nf4, "
-                          "fp4, fp8, fp16 or mixed_4bit.")
+                          "fp4, fp8, fp16 ,mixed_fp4 or mixed_fp8.")
```
Comment (hkvision) [inline python/llm/src/bigdl/llm/transformers/model.py:116]: style, the space is in the wrong place.

**Review by hkvision — APPROVED** [review]
